src/ngraph/env_tbl.hpp

```diff
@@ -0,0 +1,201 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+// This collection contains one entry for each environment variable.If an environment variable is
+// added it must be added to this list.
+//
+// In order to use this list you want to define a macro named exactly NGRAPH_DEFINE_ENVVAR
+// When you are done you should undef the macro
+
+#ifndef NGRAPH_DEFINE_ENVVAR
+#warning "NGRAPH_DEFINE_ENVVAR not defined"
+#define NGRAPH_DEFINE_ENVVAR(ENUMID, NAME, DEFAULT, DESCRIPTION)
+#endif
+
+NGRAPH_DEFINE_ENVVAR(ngraph::EnvVarEnum::NGRAPH_CODEGEN,
+                     "NGRAPH_CODEGEN",
+                     "FALSE",
+                     "Enable ngraph codegen")
+NGRAPH_DEFINE_ENVVAR(ngraph::EnvVarEnum::NGRAPH_COMPILER_DEBUGINFO_ENABLE,
+                     "NGRAPH_COMPILER_DEBUGINFO_ENABLE",
+                     "FALSE",
+                     "Enable compiler debug info when codegen is enabled")
+NGRAPH_DEFINE_ENVVAR(ngraph::EnvVarEnum::NGRAPH_COMPILER_DIAG_ENABLE,
+                     "NGRAPH_COMPILER_DIAG_ENABLE",
+                     "FALSE",
+                     "Enable compiler diagnostics when codegen is enabled")
+NGRAPH_DEFINE_ENVVAR(ngraph::EnvVarEnum::NGRAPH_COMPILER_REPORT_ENABLE,
+                     "NGRAPH_COMPILER_REPORT_ENABLE",
+                     "FALSE",
+                     "")
+NGRAPH_DEFINE_ENVVAR(ngraph::EnvVarEnum::NGRAPH_CPU_BIN_TRACER_LOG,
```

Comment:
This is an ngraph core file, not a backend file. Backend definitions belong in a backend file. This comment is for all non-ngraph-core enums